components/extension-service/src/utils/digest.ts

```diff
@@ -99,19 +100,25 @@ const getDigestFromImageAPI = async (image: string, arch: Arch) => {
             },
         });
 
-        // ! get the digest with the correct arch
+        // * get the digest with the correct arch
         if (!response.data) {
             throw new Error("No data returned from docker api");
         }
         const imageResponse: ImageResponse = response.data.find(
             (image: ImageResponse) => image.architecture === fixedArch,
         );
         if (!imageResponse) {
-            console.log(`No arch found in docker api response`);
+            // ! in this case we want to throw an Error that we can catch later, imageArchMismatchError
+            throw new Error(IMAGE_ARCH_MISMATCH_ERROR);
         }
         const digest = imageResponse.digest;
         return `${imageName}@${digest}`;
     } catch (err) {
+        // * in case of DS: Image arch mismatch, we want to throw the error
+        if (err?.message === IMAGE_ARCH_MISMATCH_ERROR) {
```

Comment:
I hate JS error handling. This is bad from a future proofing perspective, as in if we ever add more functionality to this hookpoint that also throws it's own error we'll have to keep doing this rigmarole of adding more if conditions.

A better way would be to make the try catch shorter. AFAIK, the dockerhub api errors only occur in the API request on line 97, so we should keep the error handling restricted to that

```javascript
try
{
      const response = await axios.get(url, {
          headers: {
               Authorization: `Bearer ${token}`,
          },
      });
}catch (err) {
     console.log(`Got error from docker API: `, err?.message);
     return `${imageName}:${tag}`;
}
/* rest of the code */
```
This way, for all errors other than docker hub APIs you can throw an error directly or return a value directly

Reply:
While I totally agree with the shorter try/catch blocks, in this scenario it just increases the amount of code to achieve the same effect. I have refactored the function a bit so its a little more cleaner now. 
The idea is that for all errors apart from the one where we do have a valid response, just not a matching arch; we want to revert to the initial tag & not throw an error.

![image](https://user-images.githubusercontent.com/110245015/217355635-eb64ca93-7e63-4824-b776-225e85ce92f3.png)